homeassistant/components/webhook/__init__.py

```diff
@@ -37,19 +40,30 @@ def async_register(
     webhook_id: str,
     handler: Callable[[HomeAssistant, str, Request], Awaitable[Response | None]],
     *,
-    local_only=False,
+    local_only: bool | None = False,
+    allowed_methods: Iterable[str] | None = None,
 ) -> None:
     """Register a webhook."""
     handlers = hass.data.setdefault(DOMAIN, {})
 
     if webhook_id in handlers:
         raise ValueError("Handler is already defined!")
 
+    if allowed_methods is None:
+        allowed_methods = DEFAULT_METHODS
+    allowed_methods = frozenset(method.upper() for method in allowed_methods)
```

Comment:
I feel like `upper()` in this register method should not be needed. In this case, the method can be forced/assumed to have the correct format (as it is an internal API).

For handling user input in, for example, trigger configuration, the schema can handle uppercasing the input.

Reply:
Done. I removed the `upper()` in [a6394b7](https://github.com/home-assistant/core/pull/66494/commits/a6394b717530c2ff489d3d01db9afbcdcd933315).